src/modal.js

```diff
@@ -171,6 +171,8 @@ export function setModalAsOpen(element) {
   devAssert(modalEntryStack.every((info) => info.element !== element));
   devAssert(isConnectedNode(element));
 
+  element.setAttribute('aria-modal', 'true');
```

Comment:
It looks like `aria-modal` should  be [paired with `role=dialog`](https://www.w3.org/TR/wai-aria-practices/#dialog_roles_states_props). Does the element already have `role="dialog"`? If not, there are two ways you could go about using the attribute compliantly:

- If there is an existing non-dialog role provided by the document author, leave the value as is and don't add the `aria-modal` attribute. I see there was some discussion on amp-sidebar having `role=menu` -- in the case of other components, we should be able to assume that a11y considerations are addressed within that component. Are there other types of situations where an element here would already have `role!=dialog`?
- Override any pre-existing/absent `role` value with `dialog` and always append `aria-modal`. It sounds like this is less desirable given the sidebar case.

Another question - Since we don't need to toggle the `aria-modal` attribute on open/close, can this attribute be applied once on initialization/discovery instead?

Reply:
Thanks for your insight on this @caroqliu!
Based on your feedback it feels like `aria-modal` and `role` should be set component level.
I'll close this without merging :)